jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/internal/FragmentExtension.java

```diff
@@ -107,40 +104,42 @@ public FragmentingDemandingFlusher()
         @Override
         protected boolean handle(Frame frame, Callback callback, boolean first)
         {
+            long maxFrameSize = configuration.getMaxFrameSize();
             if (first)
             {
-                if (frame.isControlFrame())
+                if (frame.isControlFrame() || maxFrameSize <= 0 || frame.getPayloadLength() <= maxFrameSize)
                 {
                     emitFrame(frame, callback);
                     return true;
                 }
+
+                // Slice the payload so we don't modify the original.
+                _payload = frame.getPayload().slice();
```

Comment:
Actually, feels "natural" that the incoming payload is consumed as it is sliced into fragments.
Other flushers, e.g. those that compress/decompress, would also consume the input, so here would feel natural too.
This is the hot path, and if we can save the allocation would be great.

Reply:
I was trying to align this class with the FragmentingFlusher which slices the payload. But yes we don't need to slice for the incoming payloads at least, so I removed this slice.

Also, this is not really the hot path, I don't think the `FragmentExtension` is actually used by anyone.